ratpack-pac4j/src/main/java/ratpack/pac4j/internal/RatpackWebContext.java

```diff
@@ -92,23 +109,43 @@ public Object getSessionAttribute(String name) {
   }
 
   @Override
+  public Object getSessionIdentifier() {
+    return session.getSession().getId();
+  }
+
+  @Override
   public String getRequestMethod() {
     return request.getMethod().getName();
   }
 
   @Override
+  public String getRemoteAddr() {
+    return request.getRemoteAddress().getHostText();
+  }
+
+  @Override
   public void writeResponseContent(String responseContent) {
     this.responseContent = responseContent;
   }
 
   @Override
   public void setResponseStatus(int code) {
-    context.getResponse().status(code);
+    response.status(code);
   }
 
   @Override
   public void setResponseHeader(String name, String value) {
-    context.getResponse().getHeaders().set(name, value);
+    response.getHeaders().set(name, value);
+  }
+
+  @Override
+  public void setResponseCharacterEncoding(String encoding) {
+    response.contentType(encoding);
```

Comment:
It's not the first time I have troubles to map this method to a framework implementation. I'll take a look on pac4j side to see if I can remove it in favor of the `setResponseContentType` method.